aten/src/ATen/native/AdaptiveMaxPooling2d.cpp

```diff
@@ -0,0 +1,362 @@
+#include "ATen/ATen.h"
+#include "ATen/NativeFunctions.h"
+#include <tuple>
+
+
+namespace at {
+namespace native {
+
+namespace {
+
+inline int start_index(int a, int b, int c) {
+  return (int)std::floor((float)(a * c) / b);
```

Comment:
That was my guess, too, but `AdaptiveAveragePooling.cpp` uses `int` and I did not want to enter a numerical analysis regarding the `float` cast. :)

Reply:
Yeah, in this case, keeping it as `int` as before is fine. This might be something to look into later, but for now we should keep it as it did before.